hal/src/nRF52840/sleep_hal.cpp

```diff
@@ -49,59 +49,106 @@
 #include "spark_wiring_vector.h"
 
 
-using namespace particle;
+using namespace particle;;
+
+namespace {
 
 typedef struct WakeupSourcePriorityCache {
-    uint32_t gpiotePriority;
-    uint32_t rtc2Priority;
-    uint32_t blePriority;
-    uint32_t lpcompPriority;
-    uint32_t usart0Priority;
-    uint32_t usart1Priority;
+    uint16_t priority;
+    uint16_t bumped;
 } WakeupSourcePriorityCache;
 
-static void bumpWakeupSourcesPriority(const hal_wakeup_source_base_t* wakeupSources, WakeupSourcePriorityCache* priority, uint32_t newPriority) {
+WakeupSourcePriorityCache gpiotePriority;
```

Comment:
These can still be all hidden within `WakeupSourcePriorityCache` and allocated on stack, as they are unused 99% of the time.